### PR TITLE
Completed Javascript Carousel Exercise - Daniel Rosenbaum

### PR DESCRIPTION
**Pull request by danielwrosenbaum**
Screen Shot 2021-02-17 at 7 25 01 PM

### CODE REVIEW

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  for (i; i < $viewList.length; i++) {
+    if ($viewList[i].className === 'view') {
+      $viewList[i].className = 'view hidden';
+      $circleButtons[i].className = ('circle far fa-circle fa-3x');
+      i++;
+      if (i < $viewList.length) {
+        $viewList[i].className = 'view';
+        $circleButtons[i].className = 'circle fas fa-circle fa-3x';
+      }
+      if (i === $viewList.length) {
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:19]: This could be an `else if`

File: javascript-carousel/main.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        $viewList[0].className = 'view';
+        $circleButtons[0].className = 'circle fas fa-circle fa-3x';
+      }
+    }
+
+  }
+}
+
+var timer = setInterval(changeImage, 3000);
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:28]: You should do this at the top of the page with your other global variables

File: javascript-carousel/main.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        $circleButtons[j].className = 'circle fas fa-circle fa-3x';
+      } else {
+        $viewList[0].className = 'view';
+        $circleButtons[0].className = 'circle fas fa-circle fa-3x';
+      }
+    }
+  }
+}
+
+$rightArrow.addEventListener('click', changeImageWithRightArrow);
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:48]: Same, do this at the top of the page

File: javascript-carousel/main.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      } else {
+        $viewList[y].className = 'view';
+        $circleButtons[y].className = 'circle fas fa-circle fa-3x';
+        x--;
+      }
+    }
+  }
+}
+
+$leftArrow.addEventListener('click', changeImageWithLeftArrow);
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:68]: Same, do this at the top of the page

File: javascript-carousel/main.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        $circleButtons[z].className = 'circle fas fa-circle fa-3x';
+      } else {
+        $viewList[z].className = 'hidden';
+        $circleButtons[z].className = 'circle far fa-circle fa-3x';
+      }
+    }
+  }
+
+}
+$navBar.addEventListener('click', changeButton);
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:94]: Same, do this at the top of the page

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great work so far!! Please see my notes

File: javascript-carousel/main.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      $viewList[i].className = 'view hidden';
+      $circleButtons[i].className = ('circle far fa-circle fa-3x');
+      i++;
+      if (i < $viewList.length) {
+        $viewList[i].className = 'view';
+        $circleButtons[i].className = 'circle fas fa-circle fa-3x';
+      } else if (i === $viewList.length) {
+        $viewList[0].className = 'view';
+        $circleButtons[0].className = 'circle fas fa-circle fa-3x';
+      }
```
Comment (thebearingedge) [inline javascript-carousel/main.js:25]: This code is repeated quite a bit. See if you can define a function that takes a new index and updates the DOM accordingly. Then, re-use that function.

```js
function activateIndex(index) {

}
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Try to think of your code as doing 3 tasks
- listen for clicks
- determine the appropriate index to activate (left or right)
- active the index

If you define functions that do those things, your code will have much less repetition in it.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**